Sources/Parsing/ParserPrinters/Pipe.swift

```diff
@@ -31,15 +31,15 @@ extension Parser {
   /// - Parameter downstream: A parser that parses the output of this parser.
   /// - Returns: A parser that pipes this parser's output into another parser.  @inlinable
   public func pipe<Downstream>(
-    @ParserBuilder<Input> _ build: () -> Downstream
+    @ParserBuilder<Downstream.Input> _ build: () -> Downstream
```

Comment:
I feel that `Output` would be slightly easier on the compiler in this position. `Self` generics are fully resolved at this staged, whereas `Downstream` aren't.
I'm a little surprised that it's working with `Downstream.Input`, because we're back in the chicken and egg situation, but I guess that the context provides enough information to resolve types here (I think because it knows that it'll returns a `Pipe` and this locks `Downstream.Input` to `Upstream.Output`).

Reply:
That makes sense. I changed it. Thank you.